ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -89,45 +87,34 @@ public DdlCommandResult run(MetaStore metaStore, boolean isValidatePhase) {
 
   private void deleteTopicIfNeeded(StructuredDataSource dataSource, boolean isValidatePhase) {
     if (!isValidatePhase && deleteTopic) {
-
-      executeWithRetries(new Callable<Void>() {
-        @Override
-        public Void call() throws Exception {
+      try {
+        ExecutorWithRetries.execute((Supplier<Future<Void>>) () -> {
           kafkaTopicClient.deleteTopics(
-              Collections.singletonList(dataSource.getKsqlTopic().getKafkaTopicName()));
-          return null;
-        }
-      }, "Could not delete the corresponding kafka topic: "
-           + dataSource.getKsqlTopic().getKafkaTopicName());
-
+              Collections.singletonList(
+                  dataSource.getKsqlTopic().getKafkaTopicName()));
+          return CompletableFuture.completedFuture(null);
+        });
+      } catch (final RuntimeException e) {
+        throw e;
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
       if (dataSource.getKsqlTopic().getKsqlTopicSerDe().getSerDe()
           == DataSource.DataSourceSerDe.AVRO) {
-        executeWithRetries(new Callable<Void>() {
-          @Override
-          public Void call() throws Exception {
-            schemaRegistryClient
-                .deleteSubject(sourceName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
-            return null;
-          }
-        }, "Could not clean up the schema registry for topic: " + sourceName);
-      }
-    }
-  }
-
-  private void executeWithRetries(Callable<Void> callable, String errorMessage) {
-    int retries = 0;
-    while (retries < NUM_RETRIES) {
-      try {
-        if (retries != 0) {
-          Thread.sleep(RETRY_BACKOFF_MS);
-        }
-        callable.call();
-        break;
-      } catch (Exception e) {
-        retries++;
-      } finally {
-        if (retries == NUM_RETRIES) {
-          throw new KsqlException(errorMessage);
+        try {
+          ExecutorWithRetries.execute((Supplier<Future<Void>>) () -> {
+            try {
+              schemaRegistryClient
```

Comment:
This won't currently retry: `schemaRegistryClient.deleteSubject` throws exceptions that aren't `RetriableException` so the code won't retry it.

Reply:
Fixed in ExecutorWithRetries